### PR TITLE
boot/boot_test.go: fix tests with multiple kernels

### PR DESCRIPTION
**Pull request by valentindavid**
Part 1 of #14282

### CODE REVIEW

File: boot/boot_test.go [hunk 1]
```diff
@@ -2408,18 +2405,23 @@ func (s *bootenv20Suite) TestMarkBootSuccessful20BootAssetsUpdateHappy(c *C) {
 
 	shimBf := bootloader.NewBootFile("", filepath.Join(dirs.SnapBootAssetsDir, "trusted", fmt.Sprintf("shim-%s", shimHash)), bootloader.RoleRecovery)
 	assetBf := bootloader.NewBootFile("", filepath.Join(dirs.SnapBootAssetsDir, "trusted", fmt.Sprintf("asset-%s", dataHash)), bootloader.RoleRecovery)
-	runKernelBf := bootloader.NewBootFile(filepath.Join(s.kern1.Filename()), "kernel.efi", bootloader.RoleRunMode)
-	recoveryKernelBf := bootloader.NewBootFile("pc-kernel_1.snap", "kernel.efi", bootloader.RoleRecovery)
+	runKernelBf := bootloader.NewBootFile(filepath.Join(s.kern1.MountFile()), "kernel.efi", bootloader.RoleRunMode)
+	recoveryKernelBf := bootloader.NewBootFile("/var/lib/snapd/seed/snaps/pc-kernel_1.snap", "kernel.efi", bootloader.RoleRecovery)
```
Comment (pedronis) [inline boot/boot_test.go:2409]: can you explain this change, I'm not entirely following it?

**Review by pedronis — COMMENTED** [review]
question

File: boot/boot_test.go (hunk 1 above)
Comment (valentindavid) [reply boot/boot_test.go:2409]: Now we get called in KernelBootFileBuilder and RecoveryKernelBootFileBuilder with a "kernelPath", which is those paths.

Comment (valentindavid) [reply boot/boot_test.go:2409]: Also those 2 variables are now only used to verify the boot chains in an assertion later. But they were removed from the input of the test.

**Review by pedronis — APPROVED** [review]
thank you

File: boot/boot_test.go (hunk 1 above)
Comment (valentindavid) [reply boot/boot_test.go]: I have moved then in all the tests I have modified.

**Pull request merged**